src/main/java/com/adobe/aem/compgenerator/web/TabBuilderServlet.java

```diff
@@ -0,0 +1,241 @@
+        ObjectMapper mapper = new ObjectMapper();
+        Message msg = new Message(true, "Tab builder getter");
+        writer.write(mapper.writeValueAsString(msg));
+        writer.close();
+    }
+
+    @Override
+    /*
+     * Handles updates to the Dialog Tab configurations
+     */
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("application/json; charset=UTF-8");
+        resp.setCharacterEncoding("UTF-8");
+        ObjectMapper mapper = new ObjectMapper();
+        File configFile = new File(CONFIG_PATH);
+        // sanity check for config file existence
+        if (!configFile.exists()) {
+            resp.setStatus(500);
+            PrintWriter writer = resp.getWriter();
+            Message msg = new Message(false, MISSING_CONFIG_MSG);
+            writer.write(mapper.writeValueAsString(msg));
+            writer.close();
+            return;
+        }
+        boolean updated = false;
+        GenerationConfig config = CommonUtils.getComponentData(configFile);
+        Options options = config.getOptions();
+        // sanity check on request payload
+        if (req.getContentLength() <= 0) {
+            resp.setStatus(500);
+        } else {
+            // get config params as json from request body
+            String body = req.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+            JsonNode reConfig = mapper.readTree(body);
+            if (reConfig.has("moveTab")) {
+                updated = true;
+                int oldIndex = reConfig.get("oldIndex").asInt();
+                int newIndex = reConfig.get("newIndex").asInt();
+                String type = reConfig.get(TAB_TYPE).asText();
+                LOG.info("Moving " + type + " dialog tabs with indexes " + oldIndex + ", " + newIndex);
+                switch (type) {
+                    case MAIN: {
+                        List<Tab> existingTabs = options.getTabProperties();
+                        Collections.swap(existingTabs, oldIndex, newIndex);
+                        options.setTabProperties(existingTabs);
+                        break;
+                    }
+                    case SHARED: {
+                        List<Tab> existingTabs = options.getSharedTabProperties();
+                        Collections.swap(existingTabs, oldIndex, newIndex);
+                        options.setSharedTabProperties(existingTabs);
+                        break;
+                    }
+                    case GLOBAL: {
+                        List<Tab> existingTabs = options.getGlobalTabProperties();
+                        Collections.swap(existingTabs, oldIndex, newIndex);
+                        options.setGlobalTabProperties(existingTabs);
+                        break;
+                    }
+                }
+            } else if (reConfig.has("removeTab")) {
+                String id = reConfig.get("id").asText();
```

Comment:
Same comment as in other servelt - if we could move the contents of these `if` clauses to functions (e.g. `removeTab()` i think it would greatly help with code readability and maintenance.

Reply:
Good idea. Will update this as well.